### PR TITLE
Fix: bt_navigator crashes on lc transitions

### PR DESCRIPTION
**Pull request by guilyx**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   |  #2847 ||
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Our Robot - but this doesn't require moving. I'll test on Steve's simulation later this week |

---

## Description of contribution in a few bullet points

* Transitioning bt_navigator to the unconfigured state crashed in haltAllActions because the XML file path was wrong. When cleaning up one of the navigators, an empty root node was accessed which crashed the node.
* If the root node is an empty pointer I considered that nothing has to be alted - tell me if you also want the transition to fail?
* When we do manage to clean up the node, if we try to configure it again the node will crash because we're redeclaring an existing parameter without checking if it exists.

## Future work that may be required in bullet points

* When calling createTreeFromFile, we may want to check if rootNode is empty and send a warning if it is.
* Might be worth it to try and transition the states of each nodes to see if there are some misbehaviors here and there, I could do that.

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_bt_navigator/src/navigators/navigate_through_poses.cpp [hunk 1]
```diff
@@ -48,12 +54,16 @@ NavigateThroughPosesNavigator::getDefaultBTFilepath(
 {
   std::string default_bt_xml_filename;
   auto node = parent_node.lock();
+
+  if (!node->has_parameter("default_nav_through_poses_bt_xml")) {
   std::string pkg_share_dir =
```
Comment (SteveMacenski) [inline nav2_bt_navigator/src/navigators/navigate_through_poses.cpp:59]: This needs indentation for a bunch of lines you added to an if statement

File: nav2_behavior_tree/src/behavior_tree_engine.cpp [hunk 2]
```diff
@@ -87,6 +87,8 @@ BehaviorTreeEngine::createTreeFromFile(
 void
 BehaviorTreeEngine::haltAllActions(BT::TreeNode * root_node)
 {
+  if (!root_node) return;
```
Comment (SteveMacenski) [inline nav2_behavior_tree/src/behavior_tree_engine.cpp:90]: Extend to use brackets / multiple lines for styling of the codebase

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
LGTM, just small  changes

**Comment by guilyx** [comment]
@SteveMacenski should be ready

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Thanks for the help!

**Pull request merged**